### PR TITLE
dbr quant: break up test class into multiple classes

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70372 dbr quant: add limited support for `torch.nn.ModuleList`
* #70349 dbr quant support for custom leaf modules, part 3/x
* #70335 dbr quant: support for custom leaf modules, part 2/x
* #70330 dbr quant: support for custom leaf modules, part 1/x
* #70329 dbr quant: unbreak case when child module not returning any outputs
* #70328 dbr quant: support functional linear without bias
* #70257 dbr quant: support dynamic linear
* #70256 dbr quant: do not attach auto_quant_state to observers
* #70247 dbr quant: stop calling eager quant convert
* **#70246 dbr quant: break up test class into multiple classes**
* #70200 dbr quant: start recording ops which are not quantizeable
* #70196 dbr quant: start reusing static quant module mappings

Summary:

Breaks up the large `TestQuantizeDBR` test case into
1. `TestQuantizeDBRIndividualOps` for testing functionality of ops
2. `TestQuantizeDBRMultipleOps` for testing non-fusion interactions between ops
3. `TestQuantizeDBR` for everything else

We may need to refactor this more in the future, but this should
unblock things for the near future.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
python test/test_quantization.py TestQuantizeDBRIndividualOps
python test/test_quantization.py TestQuantizeDBRMultipleOps
```

Differential Revision: [D33255925](https://our.internmc.facebook.com/intern/diff/D33255925)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33255925).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33255925).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33255925).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33255925).

**Review by jerryzh168 — APPROVED** [review]

File: test/quantization/dbr/test_quantize_dbr.py [hunk 1]
```diff
@@ -107,337 +107,337 @@ def _test_auto_tracing(
...
 
-
 @skipIfNoFBGEMM
-class TestQuantizeDBR(QuantizeDBRTestCase):
-    def test_fusion(self):
+class TestQuantizeDBRIndividualOps(QuantizeDBRTestCase):
+    """
+    Tests that DBR quantization covers individual ops
+    """
+    def test_conv(self):
```
Comment (jerryzh168) [inline test/quantization/dbr/test_quantize_dbr.py:115]: do we want to reuse/merge these test cases with other tools? like fx graph mode quant?

Comment (vkuzo) [reply test/quantization/dbr/test_quantize_dbr.py:115]: I feel like that would be a good task to do when we approach OSS beta, might be too early now

**Pull request closed without merging**